doc/coq.rst

```diff
@@ -0,0 +1,383 @@
+  files of the theory. ``pkg_name`` must be a valid package name.
+  
+  Note that Coq lang 1.0 will use the Coq legacy install setup, where all
+  packages share a common root namespace and install directory,
+  ``lib/coq/user-contrib/<module_prefix>``, as customary in the Make-based Coq
+  package ecosystem.
+  
+  For compatibility, we also install under the ``user-contrib`` prefix the
+  ``.cmxs`` files that appear in ``<ocaml_libraries>``,
+
+- ``<coq_flags>`` will be passed to ``coqc`` as command-line options.
+  ``:standard`` is taken from the value set in the ``(coq (flags <flags>))``
+  field in ``env`` profile. See :ref:`dune-env` for more information.
+
+- The path to the ``<ocaml_libraries>`` installed locations will be passed to
+  ``coqdep`` and ``coqc`` using Coq's ``-I`` flag; this allows for a Coq theory
+  to depend on an ML library, making it a Coq plugin.
+
+- Your Coq theory can depend on other theories by specifying them in the
+  ``<coq_theories>`` field. Dune will then pass to Coq the corresponding flags
+  for everything to compile correctly (this corresponds to the ``-Q`` flag for
+  Coq).
+  
+  As of today, we only support composition with libraries defined in the same
+  scope (i.e. under the same :ref:`dune-project` domain). This restriction will
+  be lifted in the future. Note that composition with the Coq standard library
+  is supported, but in this case the ``Coq`` prefix has been made available in a
+  qualified way, since Coq lang 0.2.
+
+- You can enable the production of Coq's native compiler object files by setting
+  ``<coq_native_mode>`` to ``native``. This will pass ``-native-compiler on`` to
+  Coq and install the corresponding object files under ``.coq-native``, when in
+  the ``release`` profile. The regular ``dev`` profile will skip native
+  compilation to make the build faster. This has been available since Coq lang
+  0.3.
+
+  Please note: support for ``native_compute`` is **experimental** and requires
+  Coq >= 8.12.1. Furthermore, dependant libraries *must* be built with the
+  ``(mode native)`` enabled. In Addition to that, Coq must be configured to
+  support native compilation. Dune will explicitly disable the generation of
+  native compilation objects when ``(mode vo)`` is enabled, irrespective of the
+  configuration of Coq. This will be improved in the future.
+
+.. _include-subdirs-coq:
+
+Recursive Qualification of Modules
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you add:
+
+.. code:: scheme
+
+    (include_subdirs qualified)
+
+to a :ref:`dune file<dune-files>`, Dune will consider all the modules in the
+directory and its subdirectories, adding a prefix to the module name in the
+usual Coq style for subdirectories. For example, file ``A/b/C.v`` will be module
+``A.b.C``.
+
+Limitations
+~~~~~~~~~~~
+
+- ``.v`` files always depend on the native OCaml version of the Coq binary and
+  its plugins, unless the natively compiled versions are not available.
```

Comment:
"unless .. not available" is a bit of a double negative.  Can you state differently?